packages/common/base/src/blockchain/types/nft.ts

```diff
@@ -0,0 +1,30 @@
+import { Blockchain } from ".";
+import { EVMBlockchain } from "./evm";
+
+export interface EVMNFT {
+    chain: EVMBlockchain;
+    contractAddress: string;
+    tokenId: string;
+}
+
+export interface SolanaNFT {
+    mintHash: string;
+    chain: "solana";
+}
+
+export interface CardanoNFT {
+    chain: "cardano";
+    assetId: string;
+}
+
+export type NFTLocator<T extends Blockchain> = `${T}:${string}${T extends EVMBlockchain ? `:${string}` : ""}`;
+
+export type NFT = SolanaNFT | EVMNFT | CardanoNFT;
+
+export type NFTOrNFTLocator =
+    | NFT
+    | NFTLocator<"solana">
+    | NFTLocator<"ethereum">
+    | NFTLocator<"polygon">
+    | NFTLocator<"bsc">
+    | NFTLocator<"cardano">;
```

Comment:
is this part actually used? I'm not convinced that it's a good abstraction and wouldn't _yet_ be proud of making it public

Reply:
it is used, required by the react sdk

Reply:
its been public for probably a year lol